Foundation/URLSession/URLSessionTask.swift

```diff
@@ -758,25 +792,65 @@ extension _ProtocolClient : URLProtocolClient {
     func urlProtocol(_ protocol: URLProtocol, didReceive challenge: URLAuthenticationChallenge) {
         guard let task = `protocol`.task else { fatalError("Received response, but there's no task.") }
         guard let session = task.session as? URLSession else { fatalError("Task not associated with URLSession.") }
+        
+        func proceed(using credential: URLCredential?) {
+            let protectionSpace = challenge.protectionSpace
+            let authScheme = protectionSpace.authenticationMethod
+
+            task.suspend()
+            
+            guard let handler = URLSessionTask.authHandler(for: authScheme) else {
+                fatalError("\(authScheme) is not supported")
```

Comment:
From the docs of `urlSession(_:didReceive:completionHandler:)`: "This method handles only the NSURLAuthenticationMethodNTLM, NSURLAuthenticationMethodNegotiate, NSURLAuthenticationMethodClientCertificate, and NSURLAuthenticationMethodServerTrust authentication types. For all other authentication schemes, the session calls only the urlSession(_:task:didReceive:completionHandler:) method."

I don't know exactly, but this might be better as a `task.resume()` and `return` to end up in that other delegate call.

Reply:
Good catch. This is replicating existing behavior, so perhaps that merits a separate patch so we can test regression separately.

Reply:
Thanks, good catch. I'll amend.

Reply:
I need to sit down and evaluate what the impact is of making that change, because we are changing current s-c-f behavior (to realign, but I may want to make a separate patch for this just so I can run tests on only that part).